tests/FSharp.Compiler.ComponentTests/ErrorMessages/TypeMismatchTests.fs

```diff
@@ -7,6 +7,52 @@ open FSharp.Test.Compiler
 
 module ``Type Mismatch`` =
 
+    module ``Different tuple lengths`` =
+        
+        [<Fact>]
+        let ``Known type on the left``() =
+            FSharp """
+let x: int * int * int = 1, ""
+let x: int * string * int = "", 1
+let x: int * int = "", "", 1
+            """
+            |> typecheck
+            |> shouldFail
+            |> withDiagnostics [
+                (Error 1, Line 2, Col 26, Line 2, Col 31,
+                 "Type mismatch. Expecting a\n    'tuple of length 3 (int * int * int)'    \nbut given a\n    'tuple of length 2'    \n")
+                (Error 1, Line 3, Col 29, Line 3, Col 34,
+                 "Type mismatch. Expecting a\n    'tuple of length 3 (int * string * int)'    \nbut given a\n    'tuple of length 2'    \n")
+                (Error 1, Line 4, Col 20, Line 4, Col 29,
+                 "Type mismatch. Expecting a\n    'tuple of length 2 (int * int)'    \nbut given a\n    'tuple of length 3'    \n")
+            ]
+            
+        [<Fact>]
+        let ``Known type on the right``() =
+            FSharp """
+let x : int * string = 1, ""
+let a, b, c = x
```

Comment:
Just wondering: what types `a`, `b`, and `c` have after a type check here? Would it be possible to make the first two values have the types from the right side and make it available via the Symbols API in FCS?

Reply:
Right now they will just be `'a 'b 'c`. But it might be possible to assign types from the right side, will have to check that.

Reply:
At the point the mismatch is detected they will all be inference variables.